nf_core/modules/module_test.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+"""
+The ModulesTest class runs the tests locally
+"""
+
+import logging
+import questionary
+import os
+import pytest
+import sys
+import rich
+
+import nf_core.utils
+from .modules_repo import ModulesRepo
+
+log = logging.getLogger(__name__)
+
+class ModulesTest(object):
+    def __init__(
+        self,
+        module_name=None,
+        no_prompts=False,
+        pytest_args="",
+    ):
+        self.module_name = module_name
+        self.no_prompts = no_prompts
+        self.pytest_args = pytest_args
+
+    def run(self):
+        """Run test steps"""
+        if not self.no_prompts:
+            log.info(
+                "[yellow]Press enter to use default values [cyan bold](shown in brackets) [yellow]or type your own responses"
+            )
+        self.check_inputs()
+        self.set_profile()
+        self.run_pytests()
+
+    def check_inputs(self):
+        """Do more complex checks about supplied flags."""
+
+        # Get the tool name if not specified
+        if self.module_name is None:
+            modules_repo = ModulesRepo()
+            modules_repo.get_modules_file_tree()
+            self.module_name = questionary.autocomplete(
+                "Tool name:",
+                choices=modules_repo.modules_avail_module_names,
+                style=nf_core.utils.nfcore_question_style,
+            ).ask()
+        self.module_dir = os.path.join("modules", *self.module_name.split("/"))
+        self.module_test_main = os.path.join("tests", "modules", *self.module_name.split("/"), "main.nf")
```

Comment:
I wonder if this kind of module structure/path handling could be done as part of `ModulesRepo` or in an util function. I think I saw a lot of such path handling in too many places.